### PR TITLE
fix(router): prioritize the --supergraph arg when APOLLO_KEY is set

### LINKED ISSUE #534: DX when `--supergraph` and Managed Federation are both enabled
**Is your feature request related to a problem? Please describe.**
I was running a `router` with an `APOLLO_KEY` and a `APOLLO_GRAPH_REF` defined, but I was also passing `--supergraph ./local-file.graphql`.  This was a bit surprising to me and I was confused for a while because my `local-file.graphql` wasn't being served but, instead, my uplink-served graph.

**Describe the solution you'd like**
I think when `--supergraph` is passed explicitly we should potentially prefer it over Uplink.  That said, at the very least, we should potentially error early or at least warn.

**Describe alternatives you've considered**
Uplink could be the override, but specifying `--supergraph` explicit is... a pretty explicit sign of intent.

### PR DESCRIPTION
**Pull request by bnjjj**
close #534

### CODE REVIEW

File: apollo-router/src/main.rs [hunk 1]
```diff
@@ -134,22 +134,39 @@ async fn rt_main() -> Result<()> {
...
+            return Err(anyhow!(
+                r#"
     💫 Apollo Router requires a supergraph to be set using '--supergraph':
 
         $ ./router --supergraph <file>`
 
+        Or using an Uplink by setting the APOLLO_KEY and APOLLO_GRAPH_REF environment variable
+
+        $ APOLLO_KEY=XXX APOLLO_GRAPH_REF=XXX ./router
+
```
Comment (abernix) [inline apollo-router/src/main.rs:169]: The term "Uplink" isn't really socialized enough yet so probably best to say "Managed Federation" here

```suggestion
        Alternatively, to retrieve the supergraph from Apollo Studio, set the APOLLO_KEY
        and APOLLO_GRAPH_REF environment variables to your graph's settings.
        
          $ APOLLO_KEY="..." APOLLO_GRAPH_REF="..." ./router
          
        For more on Apollo Studio and Managed Federation, see our documentation:
        
          https://www.apollographql.com/docs/router/managed-federation/

```

File: apollo-router/src/main.rs [hunk 2]
```diff
@@ -134,22 +134,44 @@ async fn rt_main() -> Result<()> {
...
-            opt.supergraph_path.is_some(),
-            r#"
+        _ => {
+            return Err(anyhow!(
+                r#"
     💫 Apollo Router requires a supergraph to be set using '--supergraph':
 
         $ ./router --supergraph <file>`
 
+        Alternatively, to retrieve the supergraph from Apollo Studio, set the APOLLO_KEY
```
Comment (garypen) [inline apollo-router/src/main.rs:166]: 💫 Apollo Router requires a supergraph which may be specified using either '--supergraph':
        `$ ./router --supergraph <file>`
        or, to retrieve the supergraph from Apollo Studio, set the APOLLO_KEY

**Review by garypen — APPROVED** [review]
NIT: Small language change which I hope is clearer. I'm approving anyway, since what you have is clear enough.

File: apollo-router/src/main.rs (hunk 2 above)
Comment (abernix) [reply apollo-router/src/main.rs:166]: Ah, I see what you're capturing there and completely agree with it being clarity-adding.

I'm on my phone right now, but suggest land this and we get help from DX on word-smithing this a bit.  I could even imagine that we want to hoist managed federation to the top to some degree.

Comment (garypen) [reply apollo-router/src/main.rs:166]: I still don't love my text to be honest. I think we may need some additional explanatory text which shows that "if both --supergraph and APOLLO_KEY are set, --supergraph will be used" (or something like this). How about this?:


Apollo Router requires a supergraph. If --supergraph is specified then the supergraph is read from the specified file. If not, then, as long as APOLLO_KEY and APOLLO_GRAPH_REF environment variables are set, the supergraph will be retrieved from Apollo Studio. Note: --supergraph always takes precedence.

Comment (abernix) [reply apollo-router/src/main.rs:166]: Marking this as resolved as it's now tracked via https://github.com/apollographql/router/issues/549 to track it.

**Pull request merged**